src/lp/fixed_variable.jl

```diff
@@ -1,18 +1,62 @@
+"""
+    FixedVariableRule
```

Comment:
One Q that will need to decided, though likely in this PR: IP presolve can infer fixed variables that follow a slightly different, more relaxed rule ("A variable ``x_{j}`` is fixed if the ceiling of the current lower bound and the floor of the upper bound are equal``). How should we handle this? Should this be a separate IP presolve routine? Or should this rule encompass both cases?

Reply:
The first case I can think of would be an integer variable `x` with bounds `1.5` and `2.5`, which can then be fixed to `2`.
Two possibilities:
1. Have a separate routine that tightens the bounds of integer variables (in the example above, it would set lower and upper bounds to `2`). The variable will then be eliminated just like others.
2. Within this routine, perform an additional check for integer variables

Does this cover what you had in mind? Or are there more complex situations?

Reply:
Yes, that's the case I was considering. My point is that there is a natural choice for a ``FixedVariableRule`` for IPs that is similar, but not quite equal to that for LPs which you've implemented here.

Both of your suggestions should cover what I have in mind. In this particular case, I have no real preference between the two. I'm thinking more generally that we will have to decide how separated LP and IP routines should be if they are doing the "same thing" at a high-level. Again, doesn't have to be decided here, just highlighting it for future discussion.

Reply:
I added an internal check for integer variables: if the bounds are equal but fractional, the problem is declared infeasible.

As for tightening bounds on integer variables, I prefer to have it in a separate routine.